src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/insights/BitbucketServerCodeInsightsPullRequestDecorator.java

```diff
@@ -0,0 +1,182 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class BitbucketServerCodeInsightsPullRequestDecorator implements PullRequestBuildStatusDecorator {
+
+    private static final Logger LOGGER = Loggers.get(BitbucketServerCodeInsightsPullRequestDecorator.class);
+
+    private static final List<String> OPEN_ISSUE_STATUSES =
+            Issue.STATUSES.stream().filter(s -> !Issue.STATUS_CLOSED.equals(s) && !Issue.STATUS_RESOLVED.equals(s))
+                    .collect(Collectors.toList());
+
+    private final BitbucketClient client;
+
+    public BitbucketServerCodeInsightsPullRequestDecorator(BitbucketClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public String name() {
+        return "BitbucketServer Code Insights";
+    }
+
+    public boolean isEnabled() {
+        return client.isConfigured() && client.supportsCodeInsights();
+    }
+
+    @Override
+    public void decorateQualityGateStatus(AnalysisDetails analysisDetails, UnifyConfiguration configuration) {
+        try {
+            String project = configuration.getRequiredProperty(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_PROJECT_KEY);
+
+            String repo = configuration.getRequiredProperty(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_REPOSITORY_SLUG);
+            client.createReport(project, repo,
+                    analysisDetails.getCommitSha(),
+                    toReport(analysisDetails)
+            );
+            updateAnnotations(project, repo, analysisDetails);
+        } catch (IOException e) {
+            LOGGER.error("Could not decorate pull request for project {}", analysisDetails.getAnalysisProjectKey(), e);
+        }
+    }
+
+    private CreateReportRequest toReport(AnalysisDetails analysisDetails) {
+        Map<RuleType, Long> rules = analysisDetails.getRulesCount();
+
+        List<ReportData> reportData = new ArrayList<>();
+        reportData.add(reliabilityReport(rules.get(RuleType.BUG)));
+        reportData.add(new ReportData("Code coverage", new DataValue.Percentage(newCoverage(analysisDetails))));
+        reportData.add(securityReport(rules.get(RuleType.VULNERABILITY), rules.get(RuleType.SECURITY_HOTSPOT)));
+        reportData.add(new ReportData("Duplication", new DataValue.Percentage(newDuplication(analysisDetails))));
+        reportData.add(maintainabilityReport(rules.get(RuleType.CODE_SMELL)));
+        reportData.add(new ReportData("Analysis details", new DataValue.Link("Go to SonarQube", analysisDetails.getDashboardUrl())));
+
+        return new CreateReportRequest(reportData,
+                reportDescription(analysisDetails),
+                "SonarQube",
+                "SonarQube",
+                analysisDetails.getAnalysisDate().toInstant(),
+                analysisDetails.getDashboardUrl(),
+                "https://www.sonarqube.org/favicon-152.png",
```

Comment:
suggestion deliver this image also in the static resources so that instances of SQ that might not be able to access the internet still have an icon displayed see `/common/bug.svg`

Reply:
Fixed